src/polkadot/PolkadotEngine.ts

```diff
@@ -598,7 +595,7 @@ export class PolkadotEngine extends CurrencyEngine<
   async resyncBlockchain(): Promise<void> {
     await this.killEngine()
     await this.clearBlockchainCache()
-    await this.startEngine()
+    await this.startEngine({ seenTxCheckpoint: this.seenTxCheckpoint })
```

Comment:
I see this pattern in all the `resyncBlockchain` methods, where we pass in the `{ seenTxCheckpoint: this.seenTxCheckpoint}` block to `startEngine`. Could we get rid of this somehow, maybe by having the base class just keep whatever it had before?

Reply:
The base class is keeping the state in `this.seenTxCheckpoint`, yes. 

If `startEngine` is called without a `seenTxCheckpoint`, it will set the state in the base class to undefined. So I don't see how you propose to get rid of the need for it to be an argument to this `startEngine` invocation.

Reply:
This is defunct now that we agreed to change the core API (see core PR).